rudra/data_store/bigquery/base.py

```diff
@@ -15,27 +15,51 @@
-    def __init__(self, query_job_config=None, credential_path=None):
+    def __init__(self, query_job_config=None):
         """Initialize the BigqueryBuilder object."""
-        self.original_credential_path = os.getenv('GOOGLE_APPLICATION_CREDENTIALS') \
-                               or credential_path
-
-        try:
-            json.loads(self.original_credential_path)
-            json_credentials = True
-        except Exception as e:
-            logger.error("Not JSON credentials, reverting to local env JSON file: {}".format(e))
-            json_credentials = False
-
-        if json_credentials:
-            tfile = tempfile.NamedTemporaryFile(mode='w+', delete=False)
-            tfile.write(self.original_credential_path)
-            tfile.flush()
-            tfile.seek(0)
-            self.new_credential_path = tfile.name
-            os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = self.new_credential_path
-        else:
-            self.new_credential_path = self.original_credential_path
+        logger.info("Creating BigQuery Auth Credentials")
+        gcp_type = os.getenv("GCP_TYPE", "")
+        gcp_project_id = os.getenv("GCP_PROJECT_ID", "")
+        gcp_private_key_id = os.getenv("GCP_PRIVATE_KEY_ID", "")
+        gcp_private_key = os.getenv("GCP_PRIVATE_KEY", "")
+        gcp_client_email = os.getenv("GCP_CLIENT_EMAIL", "")
+        gcp_client_id = os.getenv("GCP_CLIENT_ID", "")
+        gcp_auth_uri = os.getenv("GCP_AUTH_URI", "")
+        gcp_token_uri = os.getenv("GCP_TOKEN_URI", "")
+        gcp_auth_provider_cert_url = os.getenv("GCP_AUTH_PROVIDER_X509_CERT_URL", "")
+        gcp_client_url = os.getenv("GCP_CLIENT_X509_CERT_URL", "")
+
+        key_file_contents = \
+            """
+            {{
+              "type": "{type}",
+              "project_id": "{project_id}",
+              "private_key_id": "{private_key_id}",
+              "private_key": "{private_key}",
+              "client_email": "{client_email}",
+              "client_id": "{client_id}",
+              "auth_uri": "{auth_uri}",
+              "token_uri": "{token_uri}",
+              "auth_provider_x509_cert_url": "{auth_provider_cert_url}",
+              "client_x509_cert_url": "{client_url}"
+            }}
+            """.format(type=gcp_type,
+                       project_id=gcp_project_id,
+                       private_key_id=gcp_private_key_id,
+                       private_key=gcp_private_key,
+                       client_email=gcp_client_email,
+                       client_id=gcp_client_id,
+                       auth_uri=gcp_auth_uri,
+                       token_uri=gcp_token_uri,
+                       auth_provider_cert_url=gcp_auth_provider_cert_url,
+                       client_url=gcp_client_url)
+        logger.info('Storing BigQuery Auth Credentials')
+        tfile = tempfile.NamedTemporaryFile(mode='w+', delete=False)
+        tfile.write(key_file_contents)
+        tfile.flush()
+        tfile.seek(0)
```

Comment:
Better to use context managers for managing File I/O. Here I don't see that we have closed the file after writing to it.

Reply:
yep closed the file now and made `delete=True`